docs/src/Upsell.doc.js

```diff
@@ -251,4 +257,149 @@ function Example(props) {
   />,
 );
 
+card(
+  <Card
+    name="Upsell.Form"
+    description="Upsell.Form can be used to include form fields and a submit button within Upsell."
+  />,
+);
+
+card(
+  <PropTable
+    name="Upsell.Form"
+    id="Upsell.Form"
+    Component={Upsell?.Form}
+    props={[
+      {
+        name: 'children',
+        type: 'React.Node',
+      },
+      {
+        name: 'onSubmit',
+        type:
+          '({| event: SyntheticMouseEvent<HTMLButtonElement> | SyntheticKeyboardEvent<HTMLButtonElement> | SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement> |}) => void',
+        required: true,
+      },
+      {
+        name: 'submitButtonText',
+        type: 'string',
+        required: true,
+      },
+      {
+        name: 'submitButtonAccessibilityLabel',
+        type: 'string',
+        required: true,
+      },
+      {
+        name: 'submitButtonDisabled',
+        type: 'boolean',
+      },
+    ]}
+  />,
+);
+
+card(
+  <Example
+    id="formExample"
+    name="Example: Upsell with Form"
+    defaultCode={`
+function Example(props) {
+  const [value, setValue] = React.useState('');
+  return (
+    <Upsell
+      title="Give $30, get $60 in ads credit"
+      message="Earn $60 of ads credit, and give $30 of ads credit to a friend"
+      dismissButton={{
+        accessibilityLabel: 'Dismiss banner',
+        onDismiss: ()=>{},
+      }}
+      imageData={{
+        component: <Icon icon="pinterest" accessibilityLabel="Pin" color="darkGray" size={32}/>
+      }}
+    >
+      <Upsell.Form
+        onSubmit={(event) => {event.preventDefault();}}
+        submitButtonText="Submit"
+        submitButtonAccessibilityLabel="Submit button"
```

Comment:
We should avoid ever putting the word "button" in a label for a button (becomes repetitive for a screen reader)

Maybe `submitButtonAccessibilityLabel="Submit name for ads credit"`